homeassistant/components/light/abode.py

```diff
@@ -51,7 +51,12 @@ def turn_on(self, **kwargs):
                 *kwargs[ATTR_HS_COLOR]))
 
         if ATTR_BRIGHTNESS in kwargs and self._device.is_dimmable:
-            self._device.set_level(kwargs[ATTR_BRIGHTNESS])
+            # Convert HASS brightness (0-255) to Abode brightness (0-100)
+            brightness = int(kwargs[ATTR_BRIGHTNESS] * 100 / 255)
```

Comment:
If 99 is the upper limit the range should be mapped to 0...99.

Reply:
It's possible to pass a value of 100 to Abode but for some reason when in the response it returns 99, so AbodePy throws an error when the passed value != return value. I'm new to Python but would something like this be a better solution?

`brightness = int(max(min(kwargs[ATTR_BRIGHTNESS] * 100 / 255, 99), 0))`

Reply:
It's just math: `kwargs[ATTR_BRIGHTNESS]` provides a value between 0 and 255:

```
0 * 100 / 255 = 0
128 * 100 / 255 = int(50.19) = 50
255 * 100 / 255 = 100

vs. 

0 * 99 / 255 = 0
128 * 99 / 255 = int(49.69) = 49
255 * 99 / 255 = 99
```

Reply:
Checkout this one: https://github.com/home-assistant/home-assistant/pull/12203

It's not exactly your case but provides some support may be.

Reply:
Lol, very good point. ;)

Reply:
Could you check another edge case? Is the light turned off if you set a brightness of 0 or is 0 the lowest brightness?

Reply:
When I set the brightness all the way to 0, the light is turned off.

Reply:
Perfect!